src/contexts/CoinsContext.tsx

```diff
@@ -0,0 +1,13 @@
+import { createContext } from 'react';
+import useCoin from '../hooks/useCoins';
+import ICoinsContext from '../interfaces/ICoinsContext';
+
+export const CoinsContext = createContext<ICoinsContext>({ coins: [] });
+
+const CoinsProvider = ({ children }: any) => {
+	const [coins] = useCoin();
```

Comment:
You should use 'useCoins' instead of 'useCoin'. You have different naming for the custom hook